lib/AST/Attr.cpp

```diff
@@ -1745,30 +1748,23 @@ DerivativeAttr *DerivativeAttr::create(ASTContext &context, bool implicit,
                                        SourceLoc atLoc, SourceRange baseRange,
```

Comment:
Please break this up into two different factory initializers. Serialization has a particular mode of use of this function that no other part of the compiler needs.

Reply:
Sorry, how would you like this to be broken up exactly? Deserialization is the only users of this initializer.

Reply:
The other lazy attributes have two separate `create` functions - one for serialization, one for parser-oriented uses.